src/Moq/SetupCollection.cs

```diff
@@ -82,7 +82,7 @@ public void RemoveAllPropertyAccessorSetups()
 
 			lock (this.setups)
 			{
-				this.setups.RemoveAll(s => s is MethodSetup ms && ms.Method.IsPropertyAccessor());
+				this.setups.RemoveAll(s => s is StubbedPropertySetup || (s is MethodSetup ms && ms.Method.IsPropertyAccessor()));
```

Comment:
This line of code is getting really bad. It's even more important now to get rid of the whole method, if possible; see https://github.com/moq/moq4/pull/1203#discussion_r694890829.